### PR TITLE
feat: automod

### LINKED ISSUE #154: Auto moderation wrappers
All auto moderation wrappers is missing.

### PR DESCRIPTION
**Pull request by MaskDuck**
closes #154

### CODE REVIEW

**Comment by Lulalaby** [comment]
Learn python lol

**Comment by Lulalaby** [comment]
still wrong

**Comment by ooliver1** [comment]
Yes, you are wrong, shocking. Fix ya trailing whitespace, single quotes, type errors, lack of blank lines in wrappers :skull:

File: nextcore/http/client/wrappers/automod.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        authentication: BotAuthentication
+            Authentication info.
+        guild_id: Snowflake
+            The guild to list the auto moderation rules for.
+        wait: bool
+            Whether to wait when getting rate limited.
+        global_priority:
+            The priority of the request for the global rate-limiter.
+        bucket_priority:
+            The priority of the request for the bucket rate-limiter.
```
Comment (TAG-Epic) [inline nextcore/http/client/wrappers/automod.py:78]: You should not add the types. This is done automatically by sphinx.

File: nextcore/http/client/wrappers/automod.py [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+        )
+        r = await self._request(
+            route=route,
+            rate_limit_key=authentication.rate_limit_key,
+            headers={"Authorization": str(authentication)},
+            bucket_priority=bucket_priority,
+            global_priority=global_priority,
+            wait=wait,
+        )
+        return await r.json()  # type: ignore [no-type-return]
```
Comment (TAG-Epic) [inline nextcore/http/client/wrappers/automod.py:106]: Missing TODO comment about validating discords api

File: nextcore/http/client/wrappers/automod.py [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+            payload["trigger_metadata"] = trigger_metadata
+        if actions is not UNDEFINED:
+            payload["actions"] = actions
+        if enabled is not UNDEFINED:
+            payload["enabled"] = enabled
+        if exempt_roles is not UNDEFINED:
+            payload["exempt_roles"] = exempt_roles
+        if exempt_channels is not UNDEFINED:
+            payload["exempt_channels"] = exempt_channels
+        headers = {"Authorization": str(authentication)}
```
Comment (TAG-Epic) [inline nextcore/http/client/wrappers/automod.py:371]: Spacing here is not great. Please follow the style used by other wrappers

File: nextcore/http/client/wrappers/automod.py [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+            "PATCH",
+            "/guilds/{guild_id}/auto-moderation/rules/{rule_id}",
+            guild_id=guild_id,
+            rule_id=rule_id,
+        )
+        payload: dict[str, Any] = {}  # TODO: Use a typehint for payload
+
+        # These have different behaviour when not provided and set to None.
+        # This only adds them if they are provided (not Undefined)
+
```
Comment (TAG-Epic) [inline nextcore/http/client/wrappers/automod.py:354]: Nitpick: remove this line?

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

**Comment by TAG-Epic** [comment]
HTTP wrappers was removed in #201 so this is not needed. Thank you for the PR though!

**Pull request closed without merging**